reef: qa: Add tests to validate synced images on rbd-mirror

ajarr opened the pull request:
backport tracker: https://tracker.ceph.com/issues/64554

---

backport of https://github.com/ceph/ceph/pull/54802 and https://github.com/ceph/ceph/pull/56077
parent tracker: https://tracker.ceph.com/issues/61617

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

idryomov commented:
jenkins test api

idryomov reviewed (changes requested):
> Conflicts:
    qa/workunits/rbd/rbd_mirror_journal.sh
    - Commit https://github.com/ajarr/ceph/commit/3fd8a03887354a0c5a11d1eaae386b09242c14a9 not backported
    "qa/workunits/rbd: merge journal and snapshot test scripts"

@ajarr The resolution turns out to be incomplete. This commit also introduced `wait_for_replaying_status_in_pool_dir()` which is used in `compare_mirror_images.sh`. Need to replace that with a call to `wait_for_status_in_pool_dir()` and spell out the status.

ajarr commented:
> > Conflicts:
> > qa/workunits/rbd/rbd_mirror_journal.sh
> > - Commit [ajarr@3fd8a03](https://github.com/ajarr/ceph/commit/3fd8a03887354a0c5a11d1eaae386b09242c14a9) not backported
> > "qa/workunits/rbd: merge journal and snapshot test scripts"
> 
> @ajarr The resolution turns out to be incomplete. This commit also introduced `wait_for_replaying_status_in_pool_dir()` which is used in `compare_mirror_images.sh`. Need to replace that with a call to `wait_for_status_in_pool_dir()` and spell out the status.

Fixed. Thanks!
